frontend/src/features/admin-form/settings/components/PaymentSettingsSection/BusinessInfoSection.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  ChangeEventHandler,
+  Dispatch,
+  FocusEventHandler,
+  KeyboardEventHandler,
+  SetStateAction,
+  useCallback,
+  useRef,
+  useState,
+} from 'react'
+import { FormControl } from '@chakra-ui/react'
+
+import { FormResponseMode, StorageFormSettings } from '~shared/types'
+
+import FormLabel from '~components/FormControl/FormLabel'
+import Input from '~components/Input'
+
+import { useMutateFormSettings } from '../../mutations'
+import { useAdminFormSettings } from '../../queries'
+
+interface BusinessFieldInputProps {
+  initialValue: string
+  handleMutation: (newAddress: string) => void
+}
+const BusinessFieldInput = ({
+  initialValue,
+  handleMutation,
+}: BusinessFieldInputProps): JSX.Element => {
+  const [value, setValue] = useState(initialValue)
+
+  const inputRef = useRef<HTMLInputElement>(null)
+
+  const handleValueChange: ChangeEventHandler<HTMLInputElement> = useCallback(
+    (e) => {
+      setValue(e.target.value.trim())
```

Comment:
Was wondering if we could delay the trimming to before inputting into handle mutation

The reason for this is when we are filling in the address it becomes really unintuitive.

e.g. We will type '36 Robinson Road', however with the current trim at setState it will be '36RobinsonRoad' and we have to go back to the spaces to add them.

Reply:
Right 👍 , definitely should delay this.

Reply:
160ea4cb

Delayed to after user has clicked away from the input, trim to apply just before the value was sent as a patch to BE